### PR TITLE
working log screen

### PR DESCRIPTION
**Pull request by beckylum0216**
fixes issue #8

### CODE REVIEW

File: controllers/messages.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"github.com/perthgophers/puddle/models"
+	"html/template"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"path"
+)
+
+// locates the log and passes it
```
Comment (nii236) [inline controllers/messages.go]: Start func comments with the func name.

`// loadFile...`

File: controllers/messages.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+func loadFile() (*models.Page, error) {
+	filename := path.Join("logs", "log.txt")
+	body, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	return &models.Page{Body: body}, nil
+}
+
+// locats and renders the html file
```
Comment (nii236) [inline controllers/messages.go]: spelling (locates)

also start this with `// renderTemplate...`

Also private funcs don't need to be commented. I like them though.

File: controllers/messages.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	lp := path.Join("views", "log.html")
+	tmpl, err := template.ParseFiles(lp)
+	if err != nil {
+		log.Println(err)
+	}
+
+	tmpl.ExecuteTemplate(w, "log.html", p)
+}
+
+// puts everything together
```
Comment (nii236) [inline controllers/messages.go]: // ServeTastic...

File: models/page.go [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package models
+
+type Page struct {
+	//Title string
```
Comment (nii236) [inline models/page.go]: Remove commented code

File: models/page.go [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package models
+
+type Page struct {
```
Comment (nii236) [inline models/page.go:4]: Document exported struct

File: views/log.html [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<h1>Hello World Puddle</h1>
+
+<div>{{printf "%s" .Body}}</div>
```
Comment (nii236) [inline views/log.html]: Make sure there is a newline at the end of the file

File: views/log.html [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<h1>Hello World Puddle</h1>
```
Comment (nii236) [inline views/log.html]: This should be full HTML as it is not a partial.

So you'll want <html>, <head>, <body> etc

File: main.go [hunk 8]
```diff
@@ -33,6 +35,9 @@ func init() {
 }
 
 func main() {
+	go http.HandleFunc("/logs/log.html", controllers.ServeTastic)
```
Comment (nii236) [inline main.go]: I suspect the goroutine is not needed on this line. Not sure though.

File: controllers/messages.go [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+func ServeTastic(w http.ResponseWriter, r *http.Request) {
```
Comment (nii236) [inline controllers/messages.go:34]: Might be worth checking the method. I'm guessing return 405 if its not a GET request.

File: controllers/messages.go (hunk 1 above)
Comment (nii236) [reply controllers/messages.go]: Also "returns the log" would be more helpful than "passes it"

**Review by nii236 — CHANGES_REQUESTED** [review]
markovchain.db.lock should be added to the .gitignore

File: main.go (hunk 8 above)
Comment (nii236) [reply main.go]: Remove it and see if it still works

File: controllers/messages.go (hunk 1 above)
Comment (beckylum0216) [reply controllers/messages.go]: edited

File: controllers/messages.go (hunk 2 above)
Comment (beckylum0216) [reply controllers/messages.go]: edited

File: controllers/messages.go (hunk 3 above)
Comment (beckylum0216) [reply controllers/messages.go]: edited

File: controllers/messages.go (hunk 9 above)
Comment (beckylum0216) [reply controllers/messages.go:34]: added code to print 405 error. Tested against postman

File: models/page.go (hunk 5 above)
Comment (beckylum0216) [reply models/page.go:4]: edited

File: models/page.go (hunk 4 above)
Comment (beckylum0216) [reply models/page.go]: removed

File: views/log.html (hunk 7 above)
Comment (beckylum0216) [reply views/log.html]: Updated file to include full html tags

File: views/log.html (hunk 6 above)
Comment (beckylum0216) [reply views/log.html]: added <br/> tag

File: main.go (hunk 8 above)
Comment (beckylum0216) [reply main.go]: edited

File: views/log.html [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+</head>
+
+<body>
+	<h1>Hello World Puddle</h1>
+
+	<div>{{printf "%s" .Body}}</div>
+
+</body>
+<br/>
+</html>
```
Comment (nii236) [inline views/log.html]: Add newline to file.

Good to merge after this 👍

**Review by nii236 — APPROVED** [review]

**Pull request merged**